aspnetcore/blazor/images.md

```diff
@@ -0,0 +1,195 @@
+
+This page examines working with images within Blazor. 
+
+## Dynamically Set Image Source Using Blazor
+
+The following example demonstrates how to dynamically set the image source using Blazor. Start by creating a `images` folder in `wwwroot` and adding three (random) images.
+
+For this example, we'll assume images are named `wallpaper1.jpg`, `wallpaper2.jpg`, `wallpaper3.jpg`. Your project directory should look something like below:
+
+```
+- Pages
+- Shared
+- ...
+- wwwroot
+  - css
+  - ...
+  - images
+    - wallpaper1.jpg
+    - wallpaper2.jpg
+    - wallpaper3.jpg
+    - ...
+```
+
+Next we add the following to your `razor` file:
+
+```razor
+<img src="@ImageSource" />
+
+@code {
+  private string ImageSource { get; set; } = string.Empty;
+}
+```
+
+The `img` component's source will now be dynamically set based on the `ImageSource` property in C#.
+
+We can now add a `ShowWallpaper` method which updates the `ImageSource` based on the `images` directory path, as well as the image `id` passed in. Note, you may have to update the image path if you've named your images differently from this example.
+
+```razor
+@code {
+  private void ShowWallpaper(int id)
+  {
+      ImageSource = $"images/wallpaper{id}.jpg";
+  }
+}
+```
+
+Finally, add buttons to toggle between showing the three images:
+
+```razor
+@for (var i = 1; i <= 3; i++)
+{
+  var imageId = i;
+  <button @onclick="() => ShowWallpaper(imageId)">
+    Wallpaper @imageId
+  </button>
+}
+```
+
+Each button will call `ShowWallpaper` with the associated `imageId`.
+
+## Stream Image Data to Client
+
+Sometimes it is necessary to send the image directly to the client instead of hosting the image in a public directory. We'll now examine how this can be done using Blazor streaming interop.
+
+First we add [`@inject`](xref:mvc/views/razor#inject) directives for the following:
+
+* <xref:System.Net.Http.HttpClient?displayProperty=fullName>
+* <xref:Microsoft.JSInterop.IJSRuntime?displayProperty=fullName>
+
+Note, for Blazor Server, if you haven't already added a `HttpClient` to your services, do so now by adding `builder.Services.AddHttpClient();` in your `Program.cs` file before `builder.Build()`. For more information, see <xref:fundamentals/http-requests>.
+
+Then create an `img` component to display the image, as well as a `button` which will trigger .NET to send the image to the client.
+
+```razor
+<img id="imageUsingImageStream" />
+<button @onclick="SetImageUsingStreamingAsync">
+    Set Image Using Image Stream
+</button>
+```
+
+Next, we add a C# method that retrieves a <xref:System.IO.Stream> for the image to be displayed. This is where you may dynamically generate an image based on the specific user, or retrieve the image from storage. In this example, we retrieve the GitHub `dotnet` avatar.
+
+```razor
+@code {
+  private Stream GetImageStream()
+  {
+      return await HttpClient.GetStreamAsync("https://avatars.githubusercontent.com/u/9141961");
```

Comment:
Maybe we should use a different image source. ++ this may set a bad example as this could cause CORS related issues. Is there some prior precedent on accessing images elsewhere in the docs?

Reply:
There isn't ... it's a case-by-case basis. In the RCL topic, I favored providing guidance for fully-working examples where I tell the reader explicitly to right-click and download the images that the example uses, then they can use the code directly from the topic (cut-'n-paste) right into an app to play with the scenarios.

https://docs.microsoft.com/en-us/aspnet/core/blazor/components/class-libraries?view=aspnetcore-5.0&tabs=visual-studio#create-an-rcl-with-static-assets

We could do that here, but I'd recommend that I circle around in 2022 to set that up and just try to get this new topic merged for now.